pkg/lib/packagekit.js

```diff
@@ -482,6 +482,28 @@ export function check_uninstalled_packages(names) {
             .then(() => uninstalled);
 }
 
+/* Check a list of packages whether they are available for installation.
+ *
+ * Returns `names` filtered for available package names.
+ */
+export function check_available_packages(names) {
+    const available = [];
+
+    if (names.length === 0)
+        return Promise.resolve(available);
+
+    return cancellableTransaction("Resolve",
+                                  [Enum.FILTER_ARCH | Enum.FILTER_NOT_SOURCE, names],
+                                  null, // don't need progress, this is fast
```

Comment:
Hmm, thinking about this: this would only work if packagekit already has local package indexes. But that would fail if not: There's no refresh here, so either the refresh happens implicitly (and then takes very long) or it doesn't happen and then this would fail.

So I suppose the better approach is to do our usual "package name by distro in manifest" thing.

Reply:
Done.